src/Arcus.Tests/IPAddressRangeTests.cs

```diff
@@ -10,900 +10,900 @@
-            yield return new object[] {-1, new IPAddressRange(ipv4Slash20.Head, ipv4Slash20.Tail), new IPAddressRange(ipv4Slash16.Head, ipv4Slash16.Tail)};
-            yield return new object[] {1, new IPAddressRange(ipv6Slash64.Head, ipv6Slash64.Tail), new IPAddressRange(ipv6Slash96.Head, ipv6Slash96.Tail)};
-            yield return new object[] {-1, new IPAddressRange(ipv6Slash96.Head, ipv6Slash96.Tail), new IPAddressRange(ipv6Slash64.Head, ipv6Slash64.Tail)};
-            yield return new object[] {-1, new IPAddressRange(ipv4All.Head, ipv4All.Tail), new IPAddressRange(ipv6All.Head, ipv6All.Tail)};
-            yield return new object[] {1, new IPAddressRange(ipv6All.Head, ipv6All.Tail), new IPAddressRange(ipv4All.Head, ipv4All.Tail)};
-            yield return new object[] {-1, new IPAddressRange(ipv4Single.Head, ipv4Single.Tail), new IPAddressRange(ipv6Single.Head, ipv6Single.Tail)};
-            yield return new object[] {1, new IPAddressRange(ipv6Single.Head, ipv6Single.Tail), new IPAddressRange(ipv4Single.Head, ipv4Single.Tail)};
-        }
-
-        [Theory]
-        [MemberData(nameof(Comparison_Values))]
-        public void CompareTo_Test(int expected,
-                                   IPAddressRange left,
-                                   IPAddressRange right)
-        {
-            // Arrange
-            // Act
-            var result = left.CompareTo(right);
-
-            // Assert
-            Assert.Equal(expected, result);
-        }
-
-        [Theory]
-        [MemberData(nameof(Comparison_Values))]
-        public void Operator_Equals_Test(int expected,
-                                          IPAddressRange left,
-                                          IPAddressRange right)
-        {
-            // Arrange
-            // Act
-            var result = left == right;
-
-            // Assert
-            Assert.Equal(expected == 0, result);
-        }
-
-        [Theory]
-        [MemberData(nameof(Comparison_Values))]
-        public void Operator_NotEquals_Test(int expected,
+   public class IPAddressRangeTests
+   {
+      #region GetHashCode
+
+      [Theory]
+      [InlineData(true, "192.168.1.5", "192.168.1.100", "192.168.1.5", "192.168.1.100")]
+      [InlineData(false, "192.168.1.5", "192.168.1.100", "10.168.1.0", "10.168.1.100")]
+      [InlineData(true, "::abcd", "ff:12::abcd", "::abcd", "ff:12::abcd")]
+      [InlineData(false, "::abcd", "ff:12::abcd", "::ef", "ff:12::1234")]
+      public void GetHashCode_Test(bool expected,
+                                   string xHead,
+                                   string xTail,
+                                   string yHead,
+                                   string yTail)
+      {
+         // Arrange
+         _ = IPAddress.TryParse(xHead, out var xHeadAddress);
+         _ = IPAddress.TryParse(xTail, out var xTailAddress);
```

Comment:
Where is `_` declared? Is this some new syntax that I'm not familiar w/?

Reply:
`_` is a [C# Discard](https://docs.microsoft.com/en-us/dotnet/csharp/discards), a feature new since 7.0. It is basically a stand-in for a variable that won't be used, and holds no values. It is better to assign than to a discard than avoiding assignment as it prevents a warning from spawning.

It is also what defines the "default" case in the new C# 8 switch expressions.

Reply:
Criminy I 100% missed that. Thanks for the edification.

Reply:
This is why I write the code you approve it